frontend/src/components/LoginDialog.jsx

```diff
@@ -22,19 +30,56 @@ export default function LoginDialog() {
   };
 
   const handleClose = () => {
+    setOpen(false);
+    setRegisterVisible('none');
+    setErrorAlertVisible('none');
+  };
+
+  const handleCloseLogin = () => {
     authService.getToken(user, password).then((data) => 
       cookies.set('token', data.token, {httpOnly: false})
     );
     setOpen(false);
+    setRegisterVisible('none');
+    setErrorAlertVisible('none');
+  };
+
+  const handleCloseRegister = () => {
+    if(email.includes('@')){
+        authService.register(user, password, password2, email).then((data) => 
+          console.log(data)
+          // cookies.set('token', data.token, {httpOnly: false})
+        );
+        setOpen(false);
+        setRegisterVisible('none');
+        setErrorAlertVisible('none');
+        // TODO: nice alert
+        alert("Account has been created!.")
+    }
+    else {
+      setErrorAlertVisible('block')
+    }
+    
   };
 
+  const expandRegisterForm = () => {
+    setRegisterVisible('block')
```

Comment:
It would be more clear to set the variable to true/false, and then use it as 
isRegisterVisible ? "block" : "none"